pkg/awsconfig/testing/firewallConfig.go

```diff
@@ -0,0 +1,410 @@
+}
+var wait = 120 * time.Second
+var wait2 = 360 * time.Second
+func main() {
+
+	creds := credentials.NewStaticCredentials(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+	region := "us-east-1"
+
+
+	ec2Client := ec2.New(session.New(&aws.Config{
+		Region:      &region,
+		Credentials: creds,
+	}))
+	//create firewall client
+	firewallClient := networkfirewall.New(session.Must(session.NewSession()), &aws.Config{
+		Region:      &region,
+		Credentials: creds,
+	})
+    awsClient := Client{
+       ec2Client, 	
+	   firewallClient,
+	}
+	//Create VPC
+	Vpc := awsClient.CreateVPC("10.0.0.0/16")
+	//Create Internet Gateway
+	IG :=awsClient.CreateInternetGateway(Vpc)
+	//Create Public Subnet
+	PublicSubnet := awsClient.CreateSubnet("10.0.0.0/24", Vpc)
+	//Create Private Subnet
+	PrivateSubnet := awsClient.CreateSubnet("10.0.1.0/24", Vpc)
+	//Create Firewall Subnet
+	FirewallSubnet := awsClient.CreateSubnet("10.0.2.0/24", Vpc)
+	//Create PublicSubnet Route Table
+	PublicRT := awsClient.CreateRouteTable(Vpc, PublicSubnet)
+	//Create PrivateSubnet Route Table
+	PrivateRT := awsClient.CreateRouteTable(Vpc, PrivateSubnet)
+	//Create FirewallSubnet Route Table
+	FirewallRT := awsClient.CreateRouteTable(Vpc, FirewallSubnet)
+	//Create IGW Route Table
+	IgRT :=awsClient.CreateRouteTableIG(Vpc, IG)
+	//Create NAT Gateway
+	NatGateway := awsClient.CreateNatGateway(PublicSubnet)
+
+	//Create route 0.0.0.0/0 in PrivateRT for NatGateway
+	awsClient.CreateRoute("0.0.0.0/0", *NatGateway.NatGateway.NatGatewayId, PrivateRT)
+	fmt.Println("Successfully Created a route 0.0.0.0/0 to NatGateway in Private Subnet")
+	//Create route 0.0.0.0/0 in FirewallSubnet for IG
+	awsClient.CreateRoute("0.0.0.0/0", *IG.InternetGateway.InternetGatewayId, FirewallRT)
+	fmt.Println("Successfully Created a route 0.0.0.0/0 to IGW in Firewall Subnet")
+
+	//Create Firewall
+	Firewall := awsClient.CreateFirewall(FirewallSubnet, Vpc)
+	//Describe Firewall
+	DescribeFirewall :=awsClient.DescribeFirewall(Firewall)
+	//Create route 0.0.0.0/0 in PublicRT for FirewallEndpoint
+	awsClient.CreateRouteToFirewall("0.0.0.0/0", *DescribeFirewall.FirewallStatus.SyncStates[*FirewallSubnet.Subnet.AvailabilityZone].Attachment.EndpointId, PublicRT)
+	fmt.Println("Successfully route 0.0.0.0/0 to the Firewall Endpoint in PublicRT ")
+	//Create route 10.0.0.0/24 in IgRt to FirewallEndpoint
+	awsClient.CreateRouteToFirewall("10.0.0.0/24", *DescribeFirewall.FirewallStatus.SyncStates[*FirewallSubnet.Subnet.AvailabilityZone].Attachment.EndpointId, IgRT)
+	fmt.Println("Successfully route 10.0.0.0/24 to the Firewall Endpoint in IgRT ")
+
+}
+
+func (c Client)CreateVPC (CidrBlock string) (ec2.CreateVpcOutput){
+	VPCinput := &ec2.CreateVpcInput{
+		CidrBlock: aws.String(CidrBlock),
+	}
+
+	VPCresult, err := c.ec2Client.CreateVpc(VPCinput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
```

Comment:
Are you looking for a specific AWS error code here? If not, this isn't gaining anything.